proud-puffins/djangoProject/earlydating/signals.py

```diff
@@ -0,0 +1,19 @@
+from django.db.models.signals import post_save
+from django.contrib.auth.models import User
+from django.contrib.auth.models import Group
+from .models import Profile
+
+
+#  Listen to Users table and crete a new profile whenever new user is created
+def create_profile(sender, instance, created, **kwargs):
+    if created and not kwargs.get('raw', False):
+        try:
+            group = Group.objects.get(name='profile')
+        except Group.DoesNotExist:
```

Comment:
This is a great use of exception handling. 

I see that this function only does anything if `created and not kwargs.get('raw', False)` evaluates to True. Assuming that nothing needs to happen at all if that statement is False, this would be a good opportunity to apply early exiting.

Instead of this:
```
if some_condition:
    do_stuff()
# end of function
```

Try this:

```suggestion
if not condition:
    return
do_stuff()
# end of function
```

This way, any conditions whereby the function should end early are handled first, and your function can be less nested.